src/vgr_cairo.mli

```diff
@@ -0,0 +1,117 @@
+(*---------------------------------------------------------------------------
+   Copyright 2014 Arthur Wendling, Daniel C. Bünzli. All rights reserved.
+   Distributed under the BSD3 license, see license at the end of the file.
+   %%NAME%% release %%VERSION%%
+  ---------------------------------------------------------------------------*)
+
+(* Based on the Vgr_htmlc documentation by Daniel C. Bünzli. *)
+
+(** Vg Cairo renderer.
+
+    {b Dependencies:}
+    {ul {- {e {{:http://forge.ocamlcore.org/projects/cairo/}Cairo2 bindings
+     for OCaml}}}
+     {- {e {{:http://cairographics.org/}Cairo Graphics library}}}}
+
+    {e Release %%VERSION%% — %%MAINTAINER%% } *)
+
+(** {1:target Cairo render targets} *)
+
+val target : ?resolution:Gg.V2.t -> [< `PDF | `PNG | `PS | `SVG ] ->
+  Vg.Vgr.dst_stored Vg.Vgr.target
+(** [target resolution fmt] is a render target for rendering to the stored
+    destination given to {!Vg.Vgr.create} in the chosen format [fmt].
+    {ul
+    {- [resolution], specifies the rendering resolution in samples per
+       meters. The PDF, PS and SVG formats are measured in points by Cairo,
+       while the PNG format is in pixels. If unspecified, the default
+       conversion to points is used.}}
+
+    {b Multiple images.} Multiple images render on the target are not
+    supported. [Invalid_argument] is raised by {!Vg.Vgr.render} if multiple
+    images are rendered. *)
+
+val target_surface : ?size:Gg.size2 -> Cairo.Surface.t ->
+  [`Other] Vg.Vgr.target
+(** [target_surface size s] is a render target for rendering to the Cairo
+    surface [s].
+    {ul
+    {- The physical size of {{!Vg.Vgr.renderable}renderables} is ignored and
+       the view rectangle is mapped on the surface size.}
+    {- [size], Surfaces created with [Cairo.Surface] have a valid size, while
+       file based surfaces have a size of zero by default: If the size of the
+       surface can not be determined, the optional argument [size] is used
+       instead. [Invalid_argument] is raised if the size is invalid.}}
```

Comment:
Oh yes, sure. The cairo bindings provides five modules for creating surfaces: the generic Surface, and the file-based PDF, PS, SVG and PNG. The first one, Surface, creates and manages the surfaces in memory and is the motivation for the function `target_surface` since it provides a lot of [non-file formats](https://github.com/Chris00/ocaml-cairo/blob/master/src/cairo.mli#L1021) (and you could just use the simpler `target` if you wanted to dump the result in a file.) The function to create a new surface has the type `format -> width:int -> height:int -> surface`. While there are no resize capability, such a surface reports the correct size when asked (so the `?size` argument is ignored.)

The four remaining modules allow you to create a file-based surface. For unknown reasons, this surfaces report a size of `(0,0)` when asked (so the `?size` argument is used.) The PDF and PS surfaces do support resizing (but I think it would be confusing to arbitrarily resize for them.)

I expect people to use `target_surface` with surfaces created by the module Surface, typically for rendering textures that are then displayed directly to the screen with opengl or whatever. Very exceptionally, someone might want to use `target_surface` with one of the file-based module so that he can configure lowlevel stuff... but then we need to know the image size.